src/hdr_histogram.c

```diff
@@ -386,6 +386,64 @@ int hdr_calculate_bucket_config(
     return 0;
 }
 
+
+int hdr_calculate_bucket_config_pgss(
+        int64_t lowest_discernible_value,
+        int64_t highest_trackable_value,
+        int yb_bucket_factor,
+        struct hdr_histogram_bucket_config* cfg)
+{
+    int32_t sub_bucket_count_magnitude;
+
+    // yb change: hardcoded sig figs to 1 for memory performance, controlling bucket sizes with yb_bucket_factor
+    int significant_figures = 1;
+    if (lowest_discernible_value < 1 ||
+            significant_figures < 1 || 5 < significant_figures ||
+            lowest_discernible_value * 2 > highest_trackable_value)
+    {
+        return EINVAL;
+    }
+
+    cfg->lowest_discernible_value = lowest_discernible_value;
+    cfg->significant_figures = significant_figures;
+    cfg->highest_trackable_value = highest_trackable_value;
+
+    // yb change: override subbucket count magnitude to set subbuckets per bucket equal to yb_bucket_factor variable
+    switch(yb_bucket_factor)
+    {
+        case 8:
+        case 16:
+        case 32:
+            sub_bucket_count_magnitude = log(yb_bucket_factor)/log(2);
+            break;
+        default:
+            printf("Invalid yb_bucket_factor of %d entered, enforcing default of 16 \n", yb_bucket_factor);
```

Comment:
I'm not sure how I feel about a printf inside a library like this. If this was running inside Postgres, for example, where would this even log? Would it go to stdout?

Reply:
Some of the printfs I have had but removed here in hdr_histogram.c did end up in the tserver logs. Definitely better to standardize though and not have random printouts popping up, so will remove. For this msg specifically I haven't been able to find where it is in the log files, possibly because it runs during shared mem startup.